app/controllers/pulitzer/versions_controller.rb

```diff
@@ -10,10 +10,15 @@ def edit
   end
 
   def update
-    processing_version = Pulitzer::UpdateVersionStatus.new(@version,@status).call
-
+    if @status == "active" && @version.empty_required_content_elements?
```

Comment:
@gomayonqui this logic is right.

However, I think it might be cleaner if you put it inside Pulitzer::UpdateVersionStatus though.

You could have the initialize method do this logic and write an error to the interaction

```ruby
if @status == "active" && @version.empty_required_content_elements?
  @errors = ["It's not possible to activate a version without filling required content elements"]
end
```

Then you could add an errors accessor to the initializer:

```ruby
attr_accessor :errors
```

Then the controller could just check for errors

```ruby
status_updater = Pulitzer::UpdateVersionStatus.new(@version,@status)
if status_updater.errors.empty?
... do success logic
else
... do error logic
end
```

what do you think? it would clean up the controller and allow the interaction to handle the whole flow.